src/addressindex.h

```diff
@@ -21,20 +27,20 @@ struct CAddressUnspentKey {
     }
     template<typename Stream>
     void Serialize(Stream& s, int nType, int nVersion) const {
-        ser_writedata8(s, type);
+        ser_writedata8(s, (uint8_t)type); //uint8_t is used for backward compatibility
```

Comment:
I strongly suggest to begin avoiding C-Style casts.
I understand they're shorter to write but actually they subtle inject (at runtime) a mix of different casts (const, reinterpret) and therefore can lead to unexpected results.
When we know the type we're casting from and the type we're casting to use `static_cast`

```suggestion
        ser_writedata8(s, static_cast<uint8_t>(type)); //uint8_t is used for backward compatibility
```

Reply:
Done